arch/arm/core/cortex_m/reset.S

```diff
@@ -61,12 +63,44 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 #endif
 
     /* lock interrupts: will get unlocked when switch to main task */
-    bl	lock_interrupts
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
+    cpsid i
+#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+    movs.n r0, #_EXC_IRQ_DEFAULT_PRIO
+    msr BASEPRI, r0
+#else
+#error Unknown ARM architecture
+#endif
+
 #ifdef CONFIG_WDOG_INIT
     /* board-specific watchdog initialization is necessary */
     bl _WdogInit
 #endif
 
+#ifdef CONFIG_INIT_STACKS
+    ldr r0, =_interrupt_stack
+    ldr r1, =0xaa
+    ldr r2, =CONFIG_ISR_STACK_SIZE
+    bl memset
+#endif
+
+    /*
+     * Set PSP and use it to boot without using MSP, so that it
+     * gets set to _interrupt_stack during initialization.
+     */
+    ldr r0, =_interrupt_stack
+    ldr r1, =CONFIG_ISR_STACK_SIZE
+    adds r0, r0, r1
+    msr PSP, r0
+    movs.n r0, #2
```

Comment:
It's probably not a big deal - but you could do this as
Can you, instead, do this exactly as the C code suggested, 
`__set_CONTROL(__get_CONTROL() | CONTROL_SPSEL_Msk);`
i.e. you "or" the SPSEL bit instead of explicitly setting.

And, pls, add some inline comment, e.g. `/* __set_CONTROL(__get_CONTROL() | CONTROL_SPSEL_Msk); */` 
or simply /* CONTROL_SPSEL_Msk */

Reply:
yes, I will, Thanks